base/range.jl

```diff
@@ -777,9 +777,9 @@ end
 function intersect(r::StepRange, s::StepRange)
     if isempty(r) || isempty(s)
         return range(first(r), step=step(r), length=0)
-    elseif step(s) < 0
+    elseif step(s) < zero(step(s))
```

Comment:
Is there any value in not recomputing `step(s)` here? Perhaps it might not be free to compute it?

Reply:
Thanks for checking. I assumed that the compiler would be able to work out that it couldn't change between accesses because StepRange is immutable.